src/index.ts

```diff
@@ -174,12 +175,60 @@ export function parseServiceAccountKey(
 }
 
 /**
- * Authenticates the gcloud tool using a service account key.
+ * Check if a given credential is WIF credential configuration.
+ *
+ * @param credFile - The  WIF credential configuration.
+ * @returns boolean.
+ */
+function isWIFCredFile(credFile: string): boolean {
+  try {
+    const creds = JSON.parse(credFile);
+    return 'type' in creds && creds.type == 'external_account';
+  } catch (err) {
+    throw new SyntaxError(`Failed to parse credentials as JSON: ${err}`);
+  }
+}
+
+/**
+ * Authenticates the gcloud tool using a service account key or WIF credential configuration
+ * discovered via GOOGLE_GHA_CREDS_PATH environment variable. An optional serviceAccountKey
+ * param is supported for legacy Actions.
  *
  * @param serviceAccountKey - The service account key used for authentication.
  * @returns exit code.
  */
 export async function authenticateGcloudSDK(
+  serviceAccountKey?: string,
+  silent = true,
```

Comment:
What does `silent` do?

Reply:
Added to doc. It's exposing [this option](https://github.com/actions/toolkit/blob/e9b0746ee3e4a146656b69f5dcdd9946a17e8d5c/packages/exec/src/interfaces.ts#L13).